src/main/java/com/rapid7/client/dcerpc/msrrp/RegistryService.java

```diff
@@ -137,6 +155,16 @@ public RegistryValue getValue(final String hiveName, final String keyPath, final
         return new RegistryValue(canonicalizedValueName, response.getType(), response.getData());
     }
 
+    public byte[] getKeySecurity(final String hiveName, final String keyPath, final int securityDescriptorType)
+            throws IOException {
+        final byte[] handle = openKey(hiveName, keyPath, SYSTEM_ACCESS);
+        final int size = getKeyInfo(hiveName, keyPath).getSecurityDescriptor();
+        final BaseRegGetKeySecurityRequest request = new BaseRegGetKeySecurityRequest(handle, securityDescriptorType,
+                size);
+        final BaseRegGetKeySecurityResponse response = callExpectSuccess(request, "BaseRegGetKeySecurity");
+        return response.getRawSecurityDescriptor();
```

Comment:
Should we close the key? Doesn't look like we do it everywhere else

Reply:
@tosmun-r7 we shouldn't, the handles are cached.

Reply:
That poses a problem now that we are able to specify an access mask though. We will need to key the cache on *both* the name and the access now.

Reply:
Yea I am thinking about this too. We could force open key for security descriptor calls instead of going through openKey